analyzer_test.go

```diff
@@ -48,6 +48,66 @@ func header(header string) *goheader.Target {
 		Path: os.TempDir(),
 	}
 }
+
+func TestAnalyzer_Analyze6(t *testing.T) {
+	a := goheader.New(
+		goheader.WithTemplate("A {{ some-value }} B"),
+		goheader.WithValues(map[string]goheader.Value{
+			"SOME-VALUE": &goheader.ConstValue{
+				RawValue: "{{ some-value }}",
+			},
+		}),
+	)
+	var issue = a.Analyze(header("A {{ SOME-VALUE }} B"))
+	require.NotNil(t, issue)
+	require.Contains(t, issue.Message(), "recursion detected")
+	issue = a.Analyze(header("A {{ SOME-VALUE }} C"))
+	require.NotNil(t, issue)
+	require.Contains(t, issue.Message(), "recursion detected")
+}
+
+func TestAnalyzer_Analyze7(t *testing.T) {
+	a := goheader.New(
+		goheader.WithTemplate("A {{ some-value1 }} B"),
+		goheader.WithValues(map[string]goheader.Value{
+			"SOME-VALUE1": &goheader.ConstValue{
+				RawValue: "{{ some-value1 }}",
+			},
+			"SOME-VALUE2": &goheader.ConstValue{
+				RawValue: "{{ some-value2 }}",
+			},
+		}),
+	)
+	var issue = a.Analyze(header("A {{ SOME-VALUE }} B"))
+	require.NotNil(t, issue)
+	require.Contains(t, issue.Message(), "recursion detected")
+	issue = a.Analyze(header("A {{ SOME-VALUE }} C"))
+	require.NotNil(t, issue)
+	require.Contains(t, issue.Message(), "recursion detected")
```

Comment:
imho, this is strange and resource ineffective to detect recursion on analyze stage:
- strange, because this error belongs to options/constructor, not method
- ineffective, because users will see the error for every analyzed package vs single misconfig error